script/provider/provider.lua

```diff
@@ -141,7 +141,7 @@ m.register 'initialized'{
     ---@async
     function (_params)
         local _ <close> = progress.create(workspace.getFirstScope().uri, lang.script.WINDOW_INITIALIZING, 0.5)
-        m.updateConfig()
+        m.updateConfig((_params or {}).luarcParentUri)
```

Comment:
Because when I read the source code, I saw that the function had arguments passed elsewhere, so I used this form as well. This is the method that I feel is the least disruptive to the code

Reply:
If the argument is **used** within the function, its naming will not have a underscore prefix, just like this one: https://github.com/LuaLS/lua-language-server/blob/0703a80c0d080ef3d668c6dcfc5ca3bdaec89c20/script/provider/provider.lua#L111-L115

The reason why it is named as `_param` (with underscore prefix) is that it is **unused** in the function (before this PR ofc)

With that said, however, I am no maintainer of this repo. I just tried to help to keep a consistent coding practice in the codebase. So if maintainer accepts the change then I have no more comments. 😂 

---

> This is the method that I feel is the least disruptive to the code

👍 I totally agree with the minimal change approach when editing, and this is one of the reasons why I oppose to a coding practice of `_xxx` means unused variable in the aforementioned refactor PR: https://github.com/LuaLS/lua-language-server/pull/2865#issuecomment-2370004529

> You said `_.*` is to indicate a variable not required in the logic that follows, but what if it is required in the **future** (when codes are modified to add more features)? Do contributor need to rename the variable and remove the `_` prefix? These are all my concerns that this new coding standard will lead to **inconsistent** in near future. And inconsistency is much worse ☹️, because at that time, this `_.*` will **lose it meaning**: a variable with `_.*` prefix cannot guarantee that it is unused.

I have already expressed my concerns at that time but no one cares 🙈

Reply:
Maybe I should remove the '_' prefix?
In my work, I often use the '_' prefix to indicate that this variable has a special purpose, such as a handle to a window. This annotation is particularly useful for me, so when I first read the code here, I didn't feel that this function argument was not being used.

Reply:
> Maybe I should remove the `_` prefix?

In this repo (LuaLS), I think yes.

> In my work, I often use the `_` prefix to indicate that this variable has a special purpose

You are right that in different repos, there can be different coding practices 😄